src/gretel_trainer/benchmark/custom/datasets.py

```diff
@@ -16,6 +16,7 @@ class CustomDataset:
     row_count: int = field(init=False, repr=False)
     column_count: int = field(init=False, repr=False)
     delimiter: str
+    public: bool = field(repr=False, default=False)
```

Comment:
Do we want/need to allow users define custom datasets as public? If yes, we should add it as an optional argument to `make_dataset` (and can remove the default here).

I'm still a little 🤔 about the "public" concept in general, not sure if this is something Benchmark should offer or if it's something the client SDK should just be smarter about...

Reply:
I think it might make sense to allow it, but to be honest, I'm not sure if we should encourage it.

Can we really rely on the SDK being smart about it? We either need to _know_ that the SDK won't upload, or we have to upload it ourselves to avoid the "times #models" amplification. So either way, benchmark has to be aware of it

Reply:
> to be honest, I'm not sure if we should encourage it

I think I agree, if for no other reason than to keep the API surface area small. I'd vote for just making it a static property of the `CustomDataset` dataclass similar to what you did on `GretelDataset`.